go/vt/vtgate/vtgate.go

```diff
@@ -618,10 +563,30 @@ func (vtg *VTGate) SplitQuery(ctx context.Context, req *proto.SplitQueryRequest,
 	return nil
 }
 
+func handleExecuteError(err error, statsKey []string, query interface{}, logger *logutil.ThrottledLogger) string {
+	errStr := err.Error() + ", vtgate: " + servenv.ListeningURL.String()
+	if strings.Contains(errStr, errDupKey) {
+		infoErrors.Add("DupKey", 1)
+	} else if strings.Contains(errStr, errTxPoolFull) {
+		normalErrors.Add(statsKey, 1)
```

Comment:
We may someday want to add a .TxPoolFull to statsKey, but I think this is ok for now.